### PR TITLE
Fix/5476

### LINKED ISSUE #5476: Authentication::SAML: with_aup setting ignored on captive portal
**Describe the bug**
If you want to display an AUP to your SAML users before they registered, you can't. By default, when you create an Authentication::SAML portal module, `with_aup` is enabled. However, when you preview captive portal, you are invited to directly click on SAML redirect link.

**To Reproduce**
1. Create a SAML source
2. Create a Root portal module
3. Create an Authentication::SAML portal module with default settings (`with_aup=1`)
4. Assign Authentication::SAML portal module to Root portal module created at step 2
5. Assign your Root portal module on your connection profile
6. Preview your connection profile

=> AUP is not displayed.

**Expected behavior**
AUP displayed before SAML redirect link.

### PR DESCRIPTION
**Pull request by julsemaan**
# Description
Adjustments for SAML + OAuth for the portal modules

# Impacts
SAML + OAuth portal modules

# Issue
fixes #5476 

# Delete branch after merge
YES

# Checklist
(REQUIRED) - [yes, no or n/a]
- [ ] Document the feature
- [ ] Add unit tests
- [ ] Add acceptance tests (TestLink)

# NEWS file entries
## Enhancements
* Adjust the settings in the admin for the SAML and OAuth portal modules

## Bug Fixes
* Honor the AUP setting of the SAML portal module

### CODE REVIEW

**Review by nqb — CHANGES_REQUESTED** [review]
- `aup_template` and `landing_template` fields are missing when editing an SAML portal module.
Not sure if it's expected.

- If I set `with_aup=0` on a SAML portal module, AUP is always displayed. Tested with
```ini
[saml_portal]
source_id=saml
actions=
fields_to_save=
custom_fields=
description=saml_portal
with_aup=0
signup_template=signin.html
pid_field=email
aup_template=aup_text.html
type=Authentication::SAML
```

**Comment by nqb** [comment]
Hello @julsemaan, 

could you provide me your feedback ?

Thanks.

**Comment by julsemaan** [comment]
I tested it and it worked fine on my end before your comment.

I will have to revisit this and see where is the issue but it did work when I pushed the code

I'll also check to add the missing fields in the admin

**Comment by nqb** [comment]
Regarding issue with `with_aup=0`, I double checked and was not able to make it working.

**Comment by julsemaan** [comment]
@nqb, the with_aup is properly honored on my end

conf:
```
[custom-saml]
source_id=maml
actions=
fields_to_save=
custom_fields=
description=custom-saml
with_aup=0
signup_template=signin.html
pid_field=email
aup_template=aup_text.html
type=Authentication::SAML
```

**Comment by nqb** [comment]
I will test again.

**Comment by nqb** [comment]
It finally worked without any change on my side (except a reboot). I will test your new changes.

File: html/captive-portal/templates/saml.html [hunk 1]
```diff
@@ -1,7 +1,14 @@
...
+  <div class="c-card u-padding u-padding-top-none">
+    [% form.get_field('aup').render | none %]
   </div>
-</div>
+      
+  <script src="/common/aup-autosubmit.js"></script>
+
+  <button id="button" type="submit" name="submit" class="c-btn c-btn--primary u-1/1 c-card--hidden">
+      [% i18n('Login') %]
+  </button>
```
Comment (nqb) [inline html/captive-portal/templates/saml.html:12]: Unless I'm missing something, there is no need for this code here.

This code add an invisible button below "I accept the terms", you can click on it and you got "You must accept the terms and conditions"

Comment (julsemaan) [reply html/captive-portal/templates/saml.html:12]: That "I accept the terms" needs a target to submit the form

You can try to remove it and we'll see if it still works but AFAIK this is needed

Comment (nqb) [reply html/captive-portal/templates/saml.html:12]: You're right.

**Pull request merged**